### PR TITLE
FF patches

### PR DESCRIPTION
**Pull request by rexagod**
This PR includes fixes for bugs discovered on the consumer side while FF
testing, including:
* Don't filter out PVs based on `spec.storageClassName`, as the index is
  not recognized.
* Move SCC checks under consumer mode.
* Remove finalizer on SCC.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 1]
```diff
@@ -319,15 +319,15 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
...
 			return reconcile.Result{}, fmt.Errorf("failed to list PersistentVolumes: %s", err)
 		}
-		if len(pvList.Items) > 0 {
-			return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
-				r.storageClassClaim.Name)
+		for _, pv := range pvList.Items {
+			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
+				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
+					r.storageClassClaim.Name)
+			}
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:330]: This for-range loop will create an in memory copy of the PV objects, please use index instead:
```suggestion
		for i := range pvList.Items {
			pv := &pvList.Items[i]
			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
					r.storageClassClaim.Name)
			}
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**